app/src/main/kotlin/com/getstrm/pace/processing_platforms/snowflake/SnowflakeClient.kt

```diff
@@ -73,9 +77,16 @@ class SnowflakeClient(
         executeRequest(request)
     }
 
-    override suspend fun listTables(): List<Table> {
+    override suspend fun listTables(pageParameters: PageParameters): PagedCollection<Table> {
         val statement =
-            "select table_schema, table_name, table_type, created as create_date, last_altered as modify_date from information_schema.tables where table_type = 'BASE TABLE';"
+            """SELECT table_schema, table_name, table_type,
+               created as create_date,
+               last_altered as modify_date
+               FROM information_schema.tables where table_type = 'BASE TABLE'
+               ORDER BY table_name
+               LIMIT ${pageParameters.pageSize}
+               OFFSET ${pageParameters.skip} ;
+            """.trimMargin()
```

Comment:
I think this needs templating

Reply:
My suggestion is to create a separate ticket for this. It's not needed for the pagination, and this MR is already too big.

Reply:
I do get what you're saying in creating a new ticket for this, but postponing these kind of refactorings will eventually result in not doing them at all. We've all been there.

Reply:
Same argument as with BigQuery. With Snowflake, we can use the default dialect, which follows ANSI SQL.

Reply:
Created separate ticket https://linear.app/strmprivacy/issue/PACE-88/change-sql-strings-to-jooq-query-builders